staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -743,7 +743,7 @@ func (s *Controller) lockedUpdateLoadBalancerHosts(service *v1.Service, hosts []
 }
 
 func wantsLoadBalancer(service *v1.Service) bool {
-	return service.Spec.Type == v1.ServiceTypeLoadBalancer
+	return service.Spec.Type == v1.ServiceTypeLoadBalancer && service.Spec.LoadBalancerClassName == nil
```

Comment:
Do we need to allow clouds to indicate interest in some class names (see my questions about "internal") ?

Reply:
I'm inclined to say "yes" but for external cloud providers only. I'd like to address this in a follow-up PR because I'm not sure yet what it means for replacing existing annotations. Some cloud providers will have to support both and that's probably fine.

Reply:
I'm fine with a followup.  It seems simple enough to have a list of prefixes that should be passed-through in addition to ""